contracts/p1/RevenueTrader.sol

```diff
@@ -88,15 +121,23 @@ contract RevenueTraderP1 is TradingP1, IRevenueTrader {
             maxTradeSlippage
         );
 
-        if (launch) {
-            tryTrade(req);
-        }
+        require(launch, "trade not worth launching");
```

Comment:
Since we only check if the trade is actually worth launching and not which trade it is, would frontrunning be an issue?

(Someone could frontrun a batch auction with a dutch auction and vice-versa, maybe a DoS vector or malicious intent)

Reply:
Right. So in the BackingManager we have DoS protection from the `tradeEnd[kind] < block.timestamp + ONE_BLOCK` condition, and I think that's important. This requires there to be at least one block between auctions of the same kind. Otherwise plausibly you could run into a situation where an auction kind is not clearing, and an attacker burns ETH for a while in order to prevent recollateralization. 

In the RevenueTrader it's more complicated because the RevenueTrader _does_ run parallel auctions as long as they are for different ERC20s. We could do the same kind of DoS protection, but only if we split this out by ERC20. I think I don't see the value here, though. With the BackingManager there is some urgency to doing the trading in a timely manner, but the same isn't true of the RevenueTrader. The DoS attacker basically has to burn ETH forever, and it's not clear how they would benefit from this. 

I might be missing something, but I don't see why it would be valuable for the RevenueTrader to do the same kind of DoS protection.